client/src/components/StarRating/StarRating.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { FunctionComponent, useState } from 'react';
+
+// Types
+import type { InputProps } from 'types';
+
+// Assets
+import { IconStar } from 'assets/images/svg';
+
+// Styles
+import { Flex } from 'styles/global';
+import * as S from './StarRating.styles';
+
+interface StarRatingProps extends InputProps {
+  numOfStars?: number;
+}
+
+const StarRating: FunctionComponent<StarRatingProps> = ({
+  id,
+  numOfStars,
+  theme,
+  elementConfig,
+  ...otherProps
+}) => {
+  const [rating, setRating] = useState<number | null>(null);
+  const [ratingHovered, setRatingHovered] = useState<number | null>(null);
+
+  return (
+    <S.StarRating>
+      <Flex justifyFlexStart>
+        {[...Array(numOfStars)].map((_, i) => {
+          const ratingValue = i + 1;
+          return (
+            <S.StarWrapper
+              key={`star-${i + 1}`}
+              themeType={theme}
+              active={
+                !!(
+                  (ratingHovered && ratingValue <= ratingHovered) ||
+                  (rating && ratingValue <= (ratingHovered || rating))
+                )
+              }
+            >
+              <label htmlFor={id}>
+                <S.Star
+                  name={id}
+                  {...elementConfig}
+                  onClick={() => setRating && setRating(ratingValue)}
+                  onMouseEnter={() => setRatingHovered && setRatingHovered(ratingValue)}
+                  onMouseLeave={() => setRatingHovered && setRatingHovered(null)}
```

Comment:
why do you need `setRating &&` and `setRatingHovered && ` checks?

Reply:
Sorry, it's a leftover from when I had code relating to star-rating in `Input` component and not in a separate component as it is now.